beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -526,6 +526,7 @@ func (s *Server) GetAttesterSlashingsV2(w http.ResponseWriter, r *http.Request)
 		Version: version.String(sourceSlashings[0].Version()),
 		Data:    attBytes,
 	}
+	w.Header().Set(api.VersionHeader, version.String(sourceSlashings[0].Version()))
```

Comment:
I feel like we shouldn't be checking each slashing version on line 502 and instead take the version on the headstate. then you can set a variable for 526 / 529 based on that version. I think there's a bug here if sourSlashings returns 0 length, line 526 will panic.

Reply:
I had originally checked the version of the headstate, but to quote Radek who put it really nicely when reviewing my PR https://github.com/prysmaticlabs/prysm/pull/14479;
> The version should be based on the version of the slashing, not the state. This is because (1) the only possible slashing versions are Phase0 and Electra, and (2) the state here can be after a fork with the slashing being before the fork.

If we are worried about `sourceSlashings` being 0. I could add a check before the range and if it's len is 0 return the headstate and nil data. Or return some sort of error? The spec doesn't seem to expect an empty case scenario.

Reply:
@james-prysm is right that this will panic if there are no slashings, which is actually most of the time. Taking the head state is fine in practice, but there is a possibility that the head state will still be from the previous fork. This will happen when the block for the first slot of the new fork hasn't arrived yet or is not proposed at all. Taking the current slot is theoretically safer.

The description of the header says "The active consensus version to which the attester slashing being submitted belongs.", which means that it has to be based on the current epoch (again, using the head state is also OK in practice).

What's most important is that because now we return the version, all items in the response have to be of the same type. So if the current fork is Electra, we can only return Electra slashings. I think it's OK to ignore the other ones.